### ISSUE TITLE
occ upgrade blocked on first app update

### ISSUE THREAD

**Issue by sanpii** [nextcloud/server#27821]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. git checkout v22.0.0
2. git submodule update --init
3. ./ooc upgrade

### Expected behaviour
Upgrade my current nextcloud instance.

### Actual behaviour
Upgrading process blocked on the first app update checking:
```
$ ./occ upgrade
...
Checking for update of app calendar in appstore
```

The requested url is https://apps.nextcloud.com/api/v1/apps.json

The execution reatches the [transfer function](https://github.com/nextcloud/3rdparty/blob/9719b46b7c61239adae3649751bbc2d510e9700a/guzzlehttp/guzzle/src/Client.php#L324) in guzzle client.

I don’t have problem with wget on the server.

### Server configuration

**Operating system:**
Debian GNU/Linux 10

**Web server:**
nginx 1.14.2

**Database:**
postgresql 13

**PHP version:**
8.0

**Nextcloud version:** (see Nextcloud admin page)
22.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:**
Update

**Where did you install Nextcloud from:**
Archive downloaded from nextcloud.com

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - calendar: 2.3.0
  - cloud_federation_api: 1.4.0
  - contacts: 3.5.1
  - contactsinteraction: 1.2.0
  - dashboard: 7.1.0
  - dav: 1.17.1
  - federatedfilesharing: 1.11.0
  - federation: 1.11.0
  - files: 1.16.0
  - files_sharing: 1.13.1
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - lookup_server_connector: 1.9.0
  - mail: 1.10.0
  - oauth2: 1.9.0
  - provisioning_api: 1.11.0
  - ransomware_detection: 0.10.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - twofactor_backupcodes: 1.10.0
  - viewer: 1.5.0
  - workflowengine: 2.3.0
Disabled:
  - accessibility
  - admin_audit
  - caniupdate
  - comments
  - deck
  - encryption
  - files_external
  - files_rightclick
  - photos
  - previewgenerator
  - systemtags
  - tasks
  - testing
  - text
  - theming
  - updatenotification
  - user_ldap
  - user_status
  - weather_status
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "config_is_read_only": true,
        "trusted_domains": [
            "cloud.homecomputing.fr"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "21.0.3.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "forcessl": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 0,
        "apps_paths": [
            {
                "path": "\/home\/sanpi\/public_html\/owncloud\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/home\/sanpi\/public_html\/owncloud\/apps2",
                "url": "\/apps2",
                "writable": true
            }
        ],
        "maintenance": false,
        "secret": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "trashbin_retention_obligation": "auto",
        "app.mail.transport": "php-mail",
        "app_install_overwrite": [
            "files_rightclick"
        ],
        "overwrite.cli.url": "https:\/\/cloud.homecomputing.fr"
    },
    "apps": {
        "accessibility": {
            "types": "",
            "enabled": "no",
            "installed_version": "1.1.0"
        },
        "admin_audit": {
            "types": "logging",
            "enabled": "no",
            "installed_version": "1.5.0"
        },
        "backgroundjob": {
            "lastjob": "112683"
        },
        "bookmarks": {
            "installed_version": "0.9.1",
            "types": "",
            "enabled": "no"
        },
        "bruteforcesettings": {
            "installed_version": "1.0.2",
            "types": "",
            "enabled": "no"
        },
        "calendar": {
            "types": "",
            "installed_version": "2.3.0",
            "enabled": "yes"
        },
        "caniupdate": {
            "enabled": "no",
            "installed_version": "0.1.2",
            "types": ""
        },
        "cloud_federation_api": {
            "types": "filesystem",
            "enabled": "yes",
            "installed_version": "1.4.0"
        },
        "comments": {
            "types": "logging",
            "installed_version": "1.5.0",
            "enabled": "no"
        },
        "contacts": {
            "installed_version": "3.5.1",
            "types": "dav",
            "enabled": "yes"
        },
        "contactsinteraction": {
            "types": "dav",
            "enabled": "yes",
            "installed_version": "1.2.0"
        },
        "core": {
            "installedat": "1404840962.3206",
            "public_files": "files_sharing\/public.php",
            "remote_filesync": "files\/appinfo\/filesync.php",
            "backgroundjobs_mode": "cron",
            "shareapi_only_share_with_group_members": "yes",
            "shareapi_allow_mail_notification": "yes",
            "public_calendar": "calendar\/share.php",
            "public_caldav": "calendar\/share.php",
            "public_gallery": "gallery\/public.php",
            "repairlegacystoragesdone": "yes",
            "global_cache_gc_lastrun": "1436300102",
            "remote_files": "dav\/appinfo\/v1\/webdav.php",
            "remote_webdav": "dav\/appinfo\/v1\/webdav.php",
            "remote_dav": "dav\/appinfo\/v2\/remote.php",
            "remote_contacts": "dav\/appinfo\/v1\/carddav.php",
            "remote_carddav": "dav\/appinfo\/v1\/carddav.php",
            "remote_calendar": "dav\/appinfo\/v1\/caldav.php",
            "remote_caldav": "dav\/appinfo\/v1\/caldav.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "shareapi_allow_public_notification": "yes",
            "lastupdateResult": "[]",
            "vendor": "nextcloud",
            "moveavatarsdone": "yes",
            "previewsCleanedUp": "1",
            "installed.bundles": "[\"CoreBundle\"]",
            "public_webfinger": "social\/lib\/webfinger.php",
            "enterpriseLogoChecked": "yes",
            "theming.variables": "d41d8cd98f00b204e9800998ecf8427e",
            "lastupdatedat": "0",
            "scss.variables": "d41d8cd98f00b204e9800998ecf8427e",
            "public_host-meta": "social\/lib\/hostmeta.php",
            "oc.integritycheck.checker": "[]",
            "lastcron": "1625556612"
        },
        "dashboard": {
            "installed_version": "7.1.0",
            "enabled": "yes",
            "types": ""
        },
        "dav": {
            "types": "filesystem",
            "enabled": "yes",
            "buildCalendarSearchIndex": "yes",
            "buildCalendarReminderIndex": "yes",
            "chunks_migrated": "1",
            "regeneratedBirthdayCalendarsForYearFix": "yes",
            "installed_version": "1.17.1"
        },
        "deck": {
            "types": "dav",
            "installed_version": "1.4.2"
        },
        "direct_menu": {
            "installed_version": "0.1",
            "types": "",
            "enabled": "no"
        },
        "federatedfilesharing": {
            "types": "",
            "enabled": "yes",
            "installed_version": "1.11.0"
        },
        "federation": {
            "types": "authentication",
            "enabled": "yes",
            "installed_version": "1.11.0"
        },
        "files": {
            "types": "filesystem",
            "enabled": "yes",
            "cronjob_scan_files": "500",
            "installed_version": "1.16.0"
        },
        "files_gpxviewer": {
            "installed_version": "0.1.a",
            "types": "",
            "enabled": "no"
        },
        "files_markdown": {
            "installed_version": "2.2.0",
            "types": "",
            "enabled": "no"
        },
        "files_opds": {
            "types": "",
            "installed_version": "0.8.9",
            "enabled": "no"
        },
        "files_rightclick": {
            "installed_version": "0.11.0",
            "types": "",
            "enabled": "no"
        },
        "files_sharing": {
            "types": "filesystem",
            "enabled": "yes",
            "installed_version": "1.13.1"
        },
        "files_texteditor": {
            "types": "",
            "installed_version": "2.5.1",
            "enabled": "no"
        },
        "files_trashbin": {
            "enabled": "yes",
            "types": "filesystem,dav",
            "installed_version": "1.11.0"
        },
        "files_tree": {
            "installed_version": "0.8.2",
            "types": "",
            "enabled": "no"
        },
        "files_versions": {
            "enabled": "yes",
            "types": "filesystem,dav",
            "installed_version": "1.14.0"
        },
        "gallery": {
            "types": "",
            "enabled": "no",
            "installed_version": "18.2.0"
        },
        "keeweb": {
            "types": "",
            "installed_version": "0.4.0",
            "enabled": "no"
        },
        "lookup_server_connector": {
            "types": "authentication",
            "enabled": "yes",
            "installed_version": "1.9.0"
        },
        "mail": {
            "types": "",
            "signed": "true",
            "installed_version": "1.10.0",
            "enabled": "yes"
        },
        "music": {
            "installed_version": "0.2",
            "types": "filesystem",
            "enabled": "no"
        },
        "news": {
            "types": "",
            "installed_version": "15.0.4",
            "enabled": "no",
            "autoPurgeMinimumInterval": "60",
            "autoPurgeCount": "200",
            "maxRedirects": "10",
            "feedFetcherTimeout": "60",
            "useCronUpdates": "1",
            "exploreUrl": "",
            "updateInterval": "3600"
        },
        "notes": {
            "types": "",
            "installed_version": "2.2.0",
            "enabled": "no",
            "ocsid": "174554"
        },
        "oauth2": {
            "types": "authentication",
            "enabled": "yes",
            "installed_version": "1.9.0"
        },
        "ocDashboard": {
            "installed_version": "1.2",
            "types": "",
            "enabled": "no"
        },
        "ojsxc": {
            "installed_version": "3.4.4",
            "types": "prelogin",
            "apiSecret": "e+DLpHyBhjuBDfw9l9r04qL",
            "enabled": "no"
        },
        "passman": {
            "installed_version": "2.1.4",
            "types": "",
            "enabled": "no"
        },
        "photos": {
            "installed_version": "1.1.0",
            "types": "",
            "enabled": "no"
        },
        "previewgenerator": {
            "installed_version": "2.3.0",
            "types": "filesystem",
            "enabled": "no"
        },
        "provisioning_api": {
            "types": "prevent_group_restriction",
            "enabled": "yes",
            "installed_version": "1.11.0"
        },
        "qownnotesapi": {
            "installed_version": "16.09.0",
            "ocsid": "173817",
            "types": "",
            "enabled": "no"
        },
        "ransomware_detection": {
            "types": "",
            "installed_version": "0.10.0",
            "enabled": "yes",
            "debug": "0"
        },
        "ransomware_protection": {
            "enabled": "no",
            "installed_version": "1.1.0",
            "types": "logging"
        },
        "settings": {
            "types": "",
            "enabled": "yes",
            "installed_version": "1.3.0"
        },
        "sharebymail": {
            "types": "filesystem",
            "enabled": "yes",
            "installed_version": "1.11.0"
        },
        "social": {
            "types": "",
            "address": "https:\/\/cloud.homecomputing.fr",
            "enabled": "no",
            "installed_version": "0.2.101"
        },
        "systemtags": {
            "types": "logging",
            "enabled": "no",
            "installed_version": "1.5.0"
        },
        "tasks": {
            "types": "",
            "ocsid": "164356",
            "installed_version": "0.9.5",
            "enabled": "no"
        },
        "tasks_enhanced": {
            "installed_version": "0.2.1",
            "types": "",
            "enabled": "no"
        },
        "text": {
            "installed_version": "2.0.0",
            "types": "dav",
            "enabled": "no"
        },
        "theming": {
            "types": "logging",
            "installed_version": "1.6.0",
            "enabled": "no"
        },
        "twofactor_backupcodes": {
            "types": "",
            "enabled": "yes",
            "installed_version": "1.10.0"
        },
        "updatenotification": {
            "caniupdate": "0.1.3",
            "types": "",
            "bookmarks": "0.10.0",
            "theming": "1.4.5",
            "bruteforcesettings": "1.0.3",
            "enabled": "no",
            "notes": "2.2.0",
            "tasks": "0.9.5",
            "passman": "2.1.2",
            "mail": "0.11.0",
            "contacts": "3.0.3",
            "calendar": "1.6.4",
            "installed_version": "1.5.0",
            "files_markdown": "2.0.5"
        },
        "user_status": {
            "installed_version": "1.0.0",
            "types": "",
            "enabled": "no"
        },
        "videos": {
            "installed_version": "1.0",
            "types": "",
            "enabled": "no"
        },
        "viewer": {
            "types": "",
            "enabled": "no",
            "installed_version": "1.5.0"
        },
        "weather_status": {
            "installed_version": "1.0.0",
            "types": "",
            "enabled": "no"
        },
        "workflowengine": {
            "types": "filesystem",
            "enabled": "yes",
            "installed_version": "2.3.0"
        }
    }
}
```
</details>

**Are you using encryption:** no

**Comment by szaimen** [comment]
Please never run the code directly from github but use the published packages instead and report back if that works for you. Thanks!

**Comment by szaimen** [comment]
Thank you for taking the time to report a bug! 👍

As this seems to be a setup issue I would like to ask you to raise your question in the forums: https://help.nextcloud.com 
If you wish support with setup issues from Nextcloud GmbH we offer this as part of the Nextcloud subscription. Learn more about this at https://nextcloud.com/enterprise/

**Comment by sanpii** [comment]
> Please never run the code directly from github but use the published packages instead and report back if that works for you. Thanks!

I tryed again with the archive but the problem persists.

**Comment by tlex** [comment]
Even though this bug report was quickly closed, I want to report the same problem using the official docker image. It doesn't look like a support case, but more like a bug.

**Comment by szaimen** [comment]
Fixed with #27848

**Comment by tlex** [comment]
Thanks, I can confirm that the update was successful now. The various pages also work. Good job.
